brick/brick.go

```diff
@@ -5,13 +5,29 @@ import "fmt"
 func main() {
 	fmt.Println("This program determines if the brick can be inserted in the hole.")
 	fmt.Print("Enter the hole dimensions (2 numbers): ")
-	var a, b float64
-	fmt.Scan(&a, &b)
+	var x, y float64
+	fmt.Scan(&x, &y)
 	fmt.Print("Enter the brick dimensions (3 numbers): ")
-	var x, y, z float64
-	fmt.Scan(&x, &y, &z)
-	if a >= y && b >= z || b >= y && a >= z || a >= x && b >= z || a >= z && b >= x || a >= x && b >= y || a >= y && b >= x {
-		fmt.Println("The brick can be inserted in the hole!")
+	var a, b, c float64
+	fmt.Scan(&a, &b, &c)
+	if a > b && a > c {
+		if b*c <= x*y {
+			fmt.Println("The brick can be inserted in the hole!")
+		} else {
+			fmt.Println("The brick cannot be inserted in the hole.")
+		}
+	} else if b > a && b > c {
+		if a*c <= x*y {
+			fmt.Println("The brick can be inserted in the hole!")
+		} else {
+			fmt.Println("The brick cannot be inserted in the hole.")
+		}
+	} else if c > a && c > b {
+		if a*b <= x*y {
+			fmt.Println("The brick can be inserted in the hole!")
+		} else {
+			fmt.Println("The brick cannot be inserted in the hole.")
+		}
 	} else {
 		fmt.Println("The brick cannot be inserted in the hole.")
 	}
```

Comment:
The algorithm is not correct.

Counter example:

 * hole dimensions: 1 5
 * brick dimensions: 3 2 2

The program will output `The brick can be inserted in the hole!`, which is not true.